src/components/App.jsx

```diff
@@ -1,32 +1,44 @@
 import React, { Component } from 'react';
-import { ChangePageWrapper } from './App.Styles';
+import { Provider } from 'react-redux';
+import store from '../redux/store';
 import GlobalStyle from './GlobalStyle';
 import ErrorBoundary from './ErrorBoundary';
 import ChangePageButton from './Helper/ChangePageButton/ChangePageButton';
 import SearchPage from './Pages/SearchPage/SearchPage';
 import DetailPage from './Pages/DetailPage/DetailPage';
 
+import { ChangePageWrapper } from './App.Styles';
+
 class App extends Component {
   state = {
-    pageType: 'search'
+    pageType: 'search',
+    selectedMovieId: ''
```

Comment:
You also can move this selectedMovieId field to store for enable additional level of persistence (on refresh i will see details if previously i've closed app on it). Also you can switch to selectedMovie (not only id) to avoid additional request here.